### DISCUSSION TITLE
depth maps not appearing correctly

### DISCUSSION THREAD

**Discussion by nickponline** [NVlabs/instant-ngp#600] [Q&A] [answered]
I'm successfully running instant-ngp and getting a good result however when I switch to `Depth Map` mode I just get plain white images, I've tried the other settings for `Colorspace` and `Tonemap` Curve but still get a white / off-white image. Is this just that there is no false coloring for the depth maps?

<img width="1115" alt="Screen Shot 2022-05-11 at 1 43 27 PM" src="https://user-images.githubusercontent.com/590151/167913806-6c121b0d-3d62-4470-96c7-44c5cf296895.png">

<img width="1197" alt="Screen Shot 2022-05-11 at 1 44 27 PM" src="https://user-images.githubusercontent.com/590151/167913819-ce301416-f22f-41de-9b9c-a48a89047a79.png">

**Comment by Tom94** [comment, marked as answer]
Hi there, the depth buffer isn't normalized to the range [0,1]. Rather, it's expressed in scene units, which is why it looks white. You can reduce the exposure slider (or press Shift+E a bunch of times) to bring it into visible range. Cheers!

**Reply by nickponline** [reply]
Thanks @Tom94 I do get some structure lowering the Exposure. 

<img width="1205" alt="image" src="https://user-images.githubusercontent.com/590151/168112219-8d55d0d1-11ad-4237-bebe-d778717b0f37.png">

I guess color space just affects the gamma curve. Maybe we could visualize the depths with a color map like [Turbo](https://ai.googleblog.com/2019/08/turbo-improved-rainbow-colormap-for.html)

I'm assuming I can export these in the code though which might be better.

**Reply by Tom94** [reply]
Yeah, for using a color map, I'd recommend getting the raw depth values into Python via `testbed.render()` and then going downstream from there.

**Reply by nickponline** [reply]
@Tom94 How should `testbed.render()` be called to return the depth map instead of RGB image?

**Reply by Tom94** [reply]
You can set the render mode to depth beforehand, just like in the GUI.
```py
testbed.render_mode = ngp.Depth
```

**Reply by nickponline** [reply]
Thank you that works!

**Reply by saunair** [reply]
Hi @Tom94, after capturing the depth map, I see that the values are between 0-255. How do I get the maximum depth I can divide the values with?

**Reply by Kikihqq** [reply]
> You can set the render mode to depth beforehand, just like in the GUI.
> 
> ```python
> testbed.render_mode = ngp.Depth
> ```

Hi, I do this to render the depth map,but I just get the white / off-white image, how can I deal with it?

**Reply by zzy729425207** [reply]
@Tom94 Hi, I set testbed.render_mode to ngp.Depth and use testbed.render to render and save depth image, but I just get the white / off-white image,how can I get right depth image?

**Reply by zzy729425207** [reply]
Hi, how can I capturing the depth map? I see that the values are 0 after testbed.render(resolution[0], resolution[1], spp, True).

**Reply by Miaowei-HNU** [reply]
@zzy729425207 Hi，did you finally get the depth map to work?  I'm having the same problem now.

**Reply by zhaoxin111** [reply]
> @zzy729425207 Hi，did you finally get the depth map to work? I'm having the same problem now.

Hi, did you finally get the depth map?
